hagelslag/data/ZarrModelGrid.py

```diff
@@ -58,8 +58,10 @@ def load_data(self):
         f = s3fs.S3Map(root=path, s3=fs, check=False)
         files.append(f)
 
-        ds = xr.open_mfdataset(files, engine='zarr').load()
-        array = ds[self.variable].values.astype('float32')
+        ds = xr.open_mfdataset(files, engine='zarr', parallel=True).load()
+        arr = ds[self.variable].values.astype('float32')
```

Comment:
This call is forcing the script to download the entire HRRR run for a single variable, which could cause a significant slowdown and a lot of bandwidth use if one only wants a single forecast hour for a given task. It would be more straightforward and much cheaper to specify the appropriate forecast hours to download and pull them at this stage rather than filtering later.

Reply:
Hmmm, without a deep understanding of Zarr, I believe the entire forecast is contained within a single zarr group (per variable, per level). Thus, I don't believe it is possible to to pull individual time steps from a forecast with the way the dataset was created. I could be wrong.

Reply:
I did a timing test with loading a single forecast hour and all forecast hours for a single variable. It took about the same amount of time for both, indicating that all the data had to be downloaded and decompressed first before subsampling the slices.